paasta_tools/cli/schemas/marathon_schema.json

```diff
@@ -7,6 +7,20 @@
         "type": "object",
         "additionalProperties": false,
         "minProperties": 1,
+        "oneOf": [
+            {
+                "properties": {
+                    "healthcheck_mode": {"enum": ["tcp", "http"]}
+                }
+            },
+            {
+                "properties": {
+                    "healthcheck_mode": {"enum": ["cmd"]},
+                    "healthcheck_cmd": {"type": "string"}
+                },
+                "required": ["healthcheck_cmd"]
```

Comment:
@solarkennedy got it working with json schema. There's a few non obvious changes between v3 and v4 e.g. you have to specify required keys like this now.